src/lfo.rs

```diff
@@ -0,0 +1,150 @@
+use core::num;
+use std::{f32::consts::PI};
+
+use crate::ring_buffer::RingBuffer;
+macro_rules! assert_close {
+    ($left:expr, $right:expr, $epsilon:expr) => {{
+        let (left, right, epsilon) = ($left, $right, $epsilon);
+        assert!(
+            (left - right).abs() <= epsilon,
+            "{} is not close to {} within an epsilon of {}",
+            left,
+            right,
+            epsilon
+        );
+    }};
+}
+
+pub struct LFO{
+    frequency: f32,
+    amplitude: f32,
+    waveType: WaveType,
+    buffer: RingBuffer<f32>,
+    sample_rate: f32,
+    phase_offset: f32
+}
+pub enum LfoParams{
+    Frequency,
+    Amplitude,
+}
+
+pub enum WaveType{
+    Sine,
+    Tri,
+    Saw,
+    Square,
+}
+
+impl LFO{
+    /// Creates a new 
+    pub fn new(sample_rate_: f32, waveType_: WaveType, frequency_: f32, amplitude_: f32)->Self{
+        let buff = RingBuffer::<f32>::new(sample_rate_ as usize);
+        let mut lfo = LFO{
+            waveType: waveType_,
+            frequency: frequency_,
+            amplitude: amplitude_,
+            sample_rate: sample_rate_,
+            buffer: buff,
+            phase_offset: 0.0
+        };
+        lfo.set_frequency(frequency_);
+        lfo.set_amplitude(amplitude_); 
+        lfo.fill_waveform();
+        lfo
+    }
+    /// Reads one sample at a time from the wavetable buffer
+    pub fn pop(&mut self) -> f32 {
```

Comment:
The naming `pop` confused me at first sight, I thought it would actually pop the value out of the wavetable, would be better to differentiate the function with the same name in the ring buffer.